### PR TITLE
ContentDB: Add reason to downloads

### PR DESCRIPTION
**Pull request by rubenwardy**
This PR adds a query parameter to allow ContentDB to distinguish between new installs, dependency installs, and updates.

## To do

This PR is Ready for Review.

### CODE REVIEW

File: builtin/mainmenu/dlg_contentstore.lua [hunk 1]
```diff
@@ -407,12 +418,12 @@ function install_dialog.handle_submit(this, fields)
...
 
 	if fields.install_all then
-		queue_download(install_dialog.package)
+		queue_download(install_dialog.package, "new")
 
 		if install_dialog.will_install_deps then
 			for _, dep in pairs(install_dialog.dependencies) do
 				if not dep.is_optional and not dep.installed and dep.package then
-					queue_download(dep.package)
+					queue_download(dep.package, "dep")
```
Comment (SmallJoker) [inline builtin/mainmenu/dlg_contentstore.lua:426]: `update` is not shortened either, so for consistency this should be `"dependency"`.

**Comment by SmallJoker** [comment]
Looks good otherwise :+1:

File: builtin/mainmenu/dlg_contentstore.lua (hunk 1 above)
Comment (nerzhul) [reply builtin/mainmenu/dlg_contentstore.lua:426]: i suggest to have constants instead of hardcoding string values in the code,  this is error prone

**Review by nerzhul — CHANGES_REQUESTED** [review]

File: builtin/mainmenu/dlg_contentstore.lua (hunk 1 above)
Comment (rubenwardy) [reply builtin/mainmenu/dlg_contentstore.lua:426]: Lua isn't statically typed, so the only way I can really do this is like so:

```lua
local REASON = setmetatable({
    dependency = "dependency",
    new = "new",
	update = "update",
}, {
	__index = function(_, key)
		error("Invalid enum value: " .. key)
	end,
})
```

is this OK?

Comment (rubenwardy) [reply builtin/mainmenu/dlg_contentstore.lua]: oh wait I'm dumb, you can just use locals

**Review by SmallJoker — APPROVED** [review]
LGTM

**Pull request merged**